tests/Application.Tests/Contracts/RecentContractServiceTests.cs

```diff
@@ -1,94 +1,131 @@
-using System.Collections.ObjectModel;
 using System.Linq;
-
 using Application.Contracts;
-
 using Domain.Contracts;
-
+using Domain.Users;
 using FluentAssertions.Execution;
 
 namespace Application.Tests.Contracts;
 
 public class RecentContractServiceTests
 {
+    private const string UserId = "123";
     private readonly IRecentContractService _cut;
+    private readonly Mock<IRecentContractRepository> _mockRecentRepo;
+    private readonly Mock<IContractRepository> _mockContractRepo;
 
     public RecentContractServiceTests()
     {
-        _cut = new RecentContractService(new Collection<Contract>());
+        _mockRecentRepo = new Mock<IRecentContractRepository>();
+        _mockContractRepo = new Mock<IContractRepository>();
+        _cut = new RecentContractService(_mockRecentRepo.Object, _mockContractRepo.Object);
     }
 
     [Fact]
-    public void AddRecent_AddsContractToQueue_WhenContractNotAlreadyInQueue()
+    public void Size_ReturnsTheExpectedAmount()
     {
         // Arrange
         var contract = new Contract();
+        var contracts = new List<RecentlyViewedContract> { new(contract.Id, new User().Id) };
+        _mockRecentRepo.Setup(repo => repo.FetchRecentContracts(It.IsAny<string>()))
+            .Returns(contracts);
 
         // Act
-        _cut.Add(contract);
+        int size = _cut.Size(UserId);
 
         // Assert
-        _cut.Size().Should().Be(1);
+        size.Should().Be(1);
     }
 
     [Fact]
-    public void AddRecent_DoesNotAddContractToQueue_WhenContractAlreadyInQueue()
+    public void AddRecent_DelegatesCorrectlyToRepository()
     {
         // Arrange
         var contract = new Contract();
+        var contracts = new List<RecentlyViewedContract> { new RecentlyViewedContract() };
+        _mockRecentRepo.Setup(repo => repo.FetchRecentContracts(It.IsAny<string>()))
+            .Returns(contracts);
 
         // Act
-        _cut.Add(contract);
-        _cut.Add(contract);
+        _cut.Add(UserId, contract);
 
         // Assert
-        _cut.Size().Should().Be(1);
+        _mockRecentRepo.Verify(repo => repo.AddRecent(UserId, contract), Times.Once);
     }
 
     [Fact]
     public void AddRecent_ShouldDequeueTheOldestContract_WhenMoreQueueCapacityIsAdded()
+    {
+        // Arrange
+        var contract1 = new Contract();
+        var contracts = new List<RecentlyViewedContract>
+        {
+            new RecentlyViewedContract(),
+            new RecentlyViewedContract(),
+            new RecentlyViewedContract(),
+            new RecentlyViewedContract(),
+        };
+
+        _mockRecentRepo.Setup(repo => repo.FetchRecentContracts(It.IsAny<string>()))
+            .Returns(contracts);
+
+        // Act
+        _cut.Add(UserId, contract1);
+
+        // Assert
+        _mockRecentRepo.Verify(repo => repo.RemoveRecent(It.IsAny<RecentlyViewedContract>()), Times.Once);
+    }
+
+    [Fact]
+    public void AddRecent_ShouldNotDequeueTheOldestContract_WhenAddingAnAlreadyAddedContract()
```

Comment:
Same here as above, it should say
```suggestion
    public void AddRecent_ShouldNotRemoveAnyRecentContract_WhenAddingAnAlreadyAddedContract()
```